substrate/runtime/staking/src/lib.rs

```diff
@@ -424,7 +424,9 @@ impl<T: Trait> Module<T> {
 			Self::on_reserved_too_low(who);
 			UpdateBalanceOutcome::AccountKilled
 		} else {
+			Self::decrease_total_stake(Self::reserved_balance(who));
```

Comment:
yeah - this is one way of doing it, but it's rather inefficient. a better way might be to just check where this function can be called and ensure that they look after the increase/reduction in stake.

Reply:
I thought this way will save anyone using this function in future code from updating the TotalStake value by calling the decrease or increase function. This is the case with the similar function `on_free_too_low` which is called 2 times in the code. what do you think?

Reply:
I thought this way will save anyone using the `set_reserved_balance` function in future code from updating the TotalStake value by calling the increase or decrease function every time. Same with the `set_free_balance` which is similar and is more through the code.